src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridItemModel.cs

```diff
@@ -0,0 +1,205 @@
+            this.RaiseAndSetIfChanged(ref _isExpanded, value);
+        }
+    }
+
+    private readonly IDisposable _modelActivationDisposable;
+    protected TreeDataGridItemModel()
+    {
+        _modelActivationDisposable = WhenModelActivated(this, static (model, disposables) =>
+        {
+            // NOTE(erri120): TreeDataGrid uses `HasChildren` to show/hide the expander.
+            model.HasChildrenObservable
+                .OnUI()
+                .SubscribeWithErrorLogging(hasChildren => model.HasChildren = hasChildren)
+                .AddTo(disposables);
+
+            var serialDisposable = new SerialDisposable();
+            serialDisposable.AddTo(disposables);
+
+            // NOTE(erri120): TreeDataGrid accesses the `Children` collection only when
+            // the user opens the expander. The order is as follows:
+            // 1) TreeDataGrid checks `HasChildren` to show/hide the expander.
+            // 2) The user clicks on the visible expander icon.
+            // 3) TreeDataGrid checks `Children` first, since `HasChildren` and the `Children` collection
+            //    can be out-of-sync, TreeDataGrid will check if there are any children and possibly
+            //    hides the expander even if `HasChildren` is `true`.
+            // 4) If `Children` is non-empty, TreeDataGrid sets `IsExpanded` to `true` and adds rows to the tree.
+            // To get a lazy-initialized collection, we use the `Children` getter to introduce a side effect
+            // that will activate the subject below.
+            // Since this ordering is depended on the TreeDataGrid implementation, it's not very robust and
+            // mostly out of our control.
+            // Additionally, subscribing to `ChildrenObservable` has to return at least one item immediately,
+            // otherwise we return an empty collection.
+            model._childrenActivation
+                .DistinctUntilChanged()
+                .Subscribe((model, serialDisposable), static (isActivated, state) =>
+                {
+                    var (model, serialDisposable) = state;
+
+                    // NOTE(erri120): We always need to reset when the observable triggers.
+                    // Note that the observable we're currently in with the `DistinctUntilChanged`
+                    // gets disposed when the model is deactivated. This is important to
+                    // understand for the model and child activation/deactivation relationships.
+                    serialDisposable.Disposable = null;
+                    CleanupChildren(model._children);
+
+                    if (isActivated)
+                    {
+                        serialDisposable.Disposable = model.ChildrenObservable
+                            .OnUI()
+                            .Bind(model._children)
+                            .SubscribeWithErrorLogging();
+                    }
+                })
+                .AddTo(disposables);
```

Comment:
This breaks when the parent item is scrolled out of view, deactivating it, while the children are still in view.

The children subscription should not be disposed when the parent is deactivated.

It should only be disposed if the parent is no longer expanded, or potentially both the parent and all the children are deactivated.